src/inscription.rs

```diff
@@ -106,6 +130,16 @@ impl Inscription {
     str::from_utf8(self.content_type.as_ref()?).ok()
   }
 
+  pub(crate) fn get_parent_id(&self) -> Option<InscriptionId> {
+    if let Some(vec) = &self.parent {
+      Some(InscriptionId::load(
+        vec.clone().try_into().expect("expected a [u8; 36]"),
+      ))
+    } else {
+      None
+    }
+  }
```

Comment:
The `expect` here seems like it can break the index too easily in the case of any malformed parent data. Maybe return `None` instead of system fail?

```suggestion
  pub(crate) fn get_parent_id(&self) -> Option<InscriptionId> {
    if let Some(vec) = &self.parent {
      if let Some(vec2) = vec.clone().try_into().ok() {
        Some(InscriptionId::load(vec2))
      } else {
        None
      }
    } else {
      None
    }
  }
```

Reply:
Thanks, good catch!

Could you please create a test that demonstrates the issue and verifies it is solved? If you can, also think about test cases around different malformed inputs.

Reply:
Thanks, good catch!

Could you please create a test that demonstrates the issue and verifies it is solved? If you can, also think about test cases around different malformed inputs.